transforms/universal/filter/spark/src/filter_transform_spark.py

```diff
@@ -10,197 +10,18 @@
-        # of FilterTransformConfiguration class
-        super().__init__(config)
-        self.filter_criteria = config.get(filter_criteria_key, filter_criteria_default)
-        self.logical_operator = config.get(filter_logical_operator_key, filter_logical_operator_default)
-        self.columns_to_drop = config.get(filter_columns_to_drop_key, filter_columns_to_drop_default)
-
-    def transform(self, data: DataFrame) -> tuple[list[DataFrame], dict[str, Any]]:
-        """
-        This implementation filters the input Spark dataframe using a SQL
-        statement and returns the filtered table and execution stats
-        :param data: input Spark DataFrame
-        :return: list of output Spark DataFrames and custom statistics
-        """
-        # initialize the metadata dictionary
-        total_docs = data.count()
-        total_columns = len(data.columns)
-        metadata = {
-            "total_docs_count": total_docs,
-            "total_columns_count": total_columns,
-        }
-
-        data.createOrReplaceTempView("spark_table")
-
-        if len(self.filter_criteria) > 0:
-            # populate metadata with filtering stats for each filter criterion
-            for filter_criterion in self.filter_criteria:
-                query = f"SELECT * FROM spark_table WHERE {filter_criterion}"
-                filter_df = data.sparkSession.sql(query)
-                # filter_df = data.where(filter_criterion)
-                docs_filtered = total_docs - filter_df.count()
-                metadata[f"docs_filtered_out_by '{filter_criterion}'"] = docs_filtered
-
-            # use filtering criteria to build the SQL query for filtering
-            filter_clauses = [f"({x})" for x in self.filter_criteria]
-            where_clause = f" {self.logical_operator} ".join(filter_clauses)
-
-            # filter using 'where' function
-            try:
-                query = f"SELECT * FROM spark_table WHERE {where_clause}"
-                filtered_df = data.sparkSession.sql(query)
-                # filtered_df = data.where(where_clause)
-            except Exception as ex:
-                logger.error(f"FilterTransform::transform failed: {ex}")
-                raise ex
-        else:
-            filtered_df = data
-
-        data.sparkSession.catalog.dropTempView("spark_table")
-
-        # drop any columns requested from the final result
-        if len(self.columns_to_drop) > 0:
-            columns_to_drop = tuple(self.columns_to_drop)
-            filtered_df_cols_dropped = filtered_df.drop(*columns_to_drop)
-        else:
-            filtered_df_cols_dropped = filtered_df
-
-        # add global filter stats to metadata
-        metadata["docs_after_filter"] = filtered_df_cols_dropped.count()
-        metadata["columns_after_filter"] = len(filtered_df_cols_dropped.columns)
-
-        return [filtered_df_cols_dropped], metadata
-
-
-class FilterTransformConfiguration(TransformConfiguration):
```

Comment:
I like the ability to run a pure python transform in the spark runtime, but I was hoping this class and its use of DateFrame could be supported.

Reply:
I tried to put some info in the documentation, but here is the bottom line. DataFrame is not a local data structure, but rather a distributed one. So I am not convinced its usage applies to individual transform